packages/flash-swap/contracts/uniswap-v2/IHifiFlashUniswapV2.sol

```diff
@@ -65,6 +65,9 @@ interface IHifiFlashUniswapV2 is IUniswapV2Callee {
         uint256 underlyingAmount
     ) external view returns (uint256 collateralRepayAmount);
 
-    /// @notice Mapping between the raw address of the pair contract and the interfaced pair contract.
-    function pairs(address pair) external view returns (IUniswapV2Pair);
+    /// @notice The Uniswap V2 factory contract address.
+    function uniV2Factory() external view returns (address);
+
+    /// @notice The Uniswap V2 init code hash.
+    function uniV2InitCodeHash() external view returns (bytes32);
```

Comment:
I presume that this stands for the init code hash of the `UniswapV2Pair` contract, doesn't it?

Reply:
Yes, that's correct. The code hash depends on the `UniswapV2Pair` contract creation code.